Drift/Private/DriftBase.cpp

```diff
@@ -3066,73 +3085,109 @@ void FDriftBase::InitServerRegistration()
 {
     if (IsPreRegistered())
     {
-        InitServerInfo(cli.server_url);
+        InitServerInfo();
         return;
     }
 
-    static const int32 defaultPort = 7777;
+    static constexpr int32 DefaultPort = 7777;
 
-    FServerRegistrationPayload payload;
-    payload.placement = defaultPlacement;
-    payload.instance_name = GetInstanceName();
-    payload.ref = buildReference;
-    payload.public_ip = GetPublicIP();
-    payload.port = !cli.port.IsEmpty() && cli.port.IsNumeric() ? FCString::Atoi(*cli.port) : defaultPort;
-    payload.command_line = FCommandLine::Get();
-    payload.pid = FPlatformProcess::GetCurrentProcessId();
-    payload.status = TEXT("starting");
+    FServerRegistrationPayload Payload;
+    Payload.placement = defaultPlacement;
+    Payload.instance_name = GetInstanceName();
+    Payload.ref = buildReference;
+    Payload.public_ip = GetPublicIP();
+    Payload.port = !cli.port.IsEmpty() && cli.port.IsNumeric() ? FCString::Atoi(*cli.port) : DefaultPort;
+    Payload.command_line = FCommandLine::Get();
+    Payload.pid = FPlatformProcess::GetCurrentProcessId();
+    Payload.status = TEXT("starting");
 
-    DRIFT_LOG(Base, Log, TEXT("Registering server ip='%s', ref='%s', placement='%s'"), *payload.public_ip, *payload.ref, *payload.placement);
+    DRIFT_LOG(Base, Log, TEXT("Registering server ip='%s', ref='%s', placement='%s'"), *Payload.public_ip, *Payload.ref, *Payload.placement);
 
-    auto request = GetGameRequestManager()->Post(driftEndpoints.servers, payload);
-    request->OnResponse.BindLambda([this](ResponseContext& context, JsonDocument& doc)
+    const auto Request = GetGameRequestManager()->Post(driftEndpoints.servers, Payload);
```

Comment:
This POST would be the one to reason a bit about, but it's probably fine to retry, since if it does register a server, fails to return the response, and we register a new one, nothing will use or refer to the old one. It'll essentially always be in "starting" state, and we can ignore or clean those out later.